internal/datastore/mysql/datastore.go

```diff
@@ -0,0 +1,114 @@
+// tuple mutations and applies it to the datastore for the specified
+// namespace.
+func (mds *mysqlDatastore) WriteTuples(ctx context.Context, preconditions []*v1.Precondition, mutations []*v1.RelationshipUpdate) (datastore.Revision, error) {
+	return datastore.NoRevision, errUnimplemented
+}
+
+// DeleteRelationships deletes all Relationships that match the provided
+// filter if all preconditions are met.
+func (mds *mysqlDatastore) DeleteRelationships(ctx context.Context, preconditions []*v1.Precondition, filter *v1.RelationshipFilter) (datastore.Revision, error) {
+	return datastore.NoRevision, errUnimplemented
+}
+
+// OptimizedRevision gets a revision that will likely already be replicated
+// and will likely be shared amongst many queries.
+func (mds *mysqlDatastore) OptimizedRevision(ctx context.Context) (datastore.Revision, error) {
+	return datastore.NoRevision, errUnimplemented
+}
+
+// HeadRevision gets a revision that is guaranteed to be at least as fresh as
+// right now.
+func (mds *mysqlDatastore) HeadRevision(ctx context.Context) (datastore.Revision, error) {
+	return datastore.NoRevision, errUnimplemented
+}
+
+// Watch notifies the caller about all changes to tuples.
+//
+// All events following afterRevision will be sent to the caller.
+func (mds *mysqlDatastore) Watch(ctx context.Context, afterRevision datastore.Revision) (<-chan *datastore.RevisionChanges, <-chan error) {
+	errs := make(chan error)
+	errs <- errUnimplemented
+	return nil, errs
+}
+
+// WriteNamespace takes a proto namespace definition and persists it,
+// returning the version of the namespace that was created.
+func (mds *mysqlDatastore) WriteNamespace(ctx context.Context, newConfig *v0.NamespaceDefinition) (datastore.Revision, error) {
+	return datastore.NoRevision, errUnimplemented
+}
+
+// ReadNamespace reads a namespace definition and version and returns it, and the revision at
+// which it was created or last written, if found.
+func (mds *mysqlDatastore) ReadNamespace(ctx context.Context, nsName string, revision datastore.Revision) (ns *v0.NamespaceDefinition, lastWritten datastore.Revision, err error) {
+	return nil, datastore.NoRevision, errUnimplemented
+}
+
+// DeleteNamespace deletes a namespace and any associated tuples.
+func (mds *mysqlDatastore) DeleteNamespace(ctx context.Context, nsName string) (datastore.Revision, error) {
+	return datastore.NoRevision, errUnimplemented
+}
+
+// ListNamespaces lists all namespaces defined.
+func (mds *mysqlDatastore) ListNamespaces(ctx context.Context, revision datastore.Revision) ([]*v0.NamespaceDefinition, error) {
+	return nil, errUnimplemented
+}
+
+// IsReady returns whether the datastore is ready to accept data. Datastores that require
+// database schema creation will return false until the migrations have been run to create
+// the necessary tables.
+func (mds *mysqlDatastore) IsReady(ctx context.Context) (bool, error) {
+	return false, errUnimplemented
+}
+
```

Comment:
There's one more datastore method now; [`NamespaceCacheKey`](https://github.com/authzed/spicedb/blob/7c81b9c100aaddfe14781b125a627f4f745e1d1a/internal/datastore/datastore.go#L69). I know there's no implementation in this PR yet but it should be the same as postgres/crdb when we get there.

Reply:
^ this is the cause of the current CI failures